src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -337,14 +337,20 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
                     }
                 }
             }
+            else if (!isTemporaryEntry(lk) &&
+                     ltx.loadWithoutRecord(lk, /*loadExpiredEntry=*/true))
+            {
+                // Cannot access an expired entry
+                return false;
+            }
             metrics.noteReadEntry(isCodeKey(lk), keySize, entrySize);
```

Comment:
I think we actually need to eagerly check `if (resources.readBytes < metrics.mLedgerReadByte)` condition on every iteration. Otherwise it's possible to read way more data than expected, which might be an issue at ledger level. You'll need to return the error from this function though.

Reply:
Good point. Fixed.